src/content/BatchRequestContent.ts

```diff
@@ -198,9 +198,25 @@ export class BatchRequestContent {
 		const requestData: RequestData = {
 			url: "",
 		};
-		const hasHttpRegex = new RegExp("^https?://");
+
 		// Stripping off hostname, port and url scheme
-		requestData.url = hasHttpRegex.test(request.url) ? "/" + request.url.split(/.*?\/\/.*?\//)[1] : request.url;
+		const hasHttpRegex = new RegExp("^https?://");
+
+		if (hasHttpRegex.test(request.url)) {
+			// Strip away host segment
+			let path = request.url.split(/.*?\/\/.*?\//)[1];
+
+			// Strip away version
+			const endOfVersionStrPos = path.indexOf("/");
+			if (endOfVersionStrPos !== -1) {
+				path = path.substring(endOfVersionStrPos, path.length);
+			}
+
+			requestData.url = path;
+		} else {
+			requestData.url = request.url;
+		}
```

Comment:
```suggestion
		requestData.url = request.url.replace(/^(?:http)?s?:?(?:\/\/)?[^\/]+\/(?:v1.0|beta)?/i, ''); // replaces <scheme>?<?>?<//><hostname:port>+</>?<version>+ by an empty string
```

What do you think of this instead? fewer allocs and conditional blocks

Reply:
That looks good!